citrination_client/client.py

```diff
@@ -271,19 +271,78 @@ def upload_file(self, file_path, data_set_id, root_path=None):
                 message = {"message": "Upload failed.",
                                    "status": r.status_code}
                 return json.dumps(message)
+        else:
+            return json.dumps({
+                    "message": "No file at specified path {0}".format(source_path)
+                })
+
+    def upload_file(self, file_path, dataset_id, root_path=None):
+        """
+        Upload file to Citrination.
+        :param file_path: File path to upload.
+        :param data_set_id: The dataset id to upload the file to.
+        :return: Response object or return code if the file was not uploaded.
+        """
+        return self.upload(dataset_id, file_path)
+
+    def list(self, dataset_id, glob=".", is_dir=False):
+        """
+        List matched filenames in a dataset on Citrination.
+        :param dataset_id: The ID of the dataset to search for files.
+        :param glob: A pattern which will be matched against files in the dataset.
+        :param is_dir: A boolean indicating whether or not the pattern should match against the beginning of paths in the dataset.
+        :return: Response object or return code if the file was not uploaded.
+        """
+        url = self.api_url + '/datasets/' + str(dataset_id) + '/list_filepaths'
```

Comment:
Related to point just below this, I don't know what the conventions are elsewhere in the URLs on citrination. I assume underscores are standard, but please make sure.